modules/dom_notifications_firebase/dom_notifications_firebase.module

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @file
+ * Contains dom_notifications_firebase.module.
+ */
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ */
+function dom_notifications_firebase_dom_notification_presave(EntityInterface $entity) {
+  /** @var \Drupal\dom_notifications\Entity\DomNotification $entity */
+  $operations = [];
+  foreach ($entity->getRecipients() as $recipient) {
+    $operations[] = [
+      'dom_notifications_firebase_send_notification',
+      [$recipient, $entity],
+    ];
+  }
+
+  $batch = [
+    'operations' => $operations,
+  ];
+
+  // Set & run batch in background.
+  batch_set($batch);
+  $batch = &batch_get();
+  $batch['progressive'] = FALSE;
+  batch_process();
+}
+
+/**
+ * Sent notification to the recipient.
+ */
+function dom_notifications_firebase_send_notification($recipient, $entity) {
```

Comment:
$recipient may have muted notifications from the channel, please check whether we should send push at all using `$entity->getChannel()->getAlertsStatus($recipient->id())`